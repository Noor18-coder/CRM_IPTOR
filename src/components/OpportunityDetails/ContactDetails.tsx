import React from 'react';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import { Card, Image, Accordion } from 'react-bootstrap';
import { DeleteCustomerContactParams, OpportunityContact } from '../../helpers/Api/models';
import OpportunityDetailsApi from '../../helpers/Api/OpportunityDetailsApi';
import ImageConfig from '../../config/ImageConfig';
import * as models from '../../helpers/Api/models';
import { AppState } from '../../store';
import { Attributes } from '../../helpers/Api';

import { deleteContactFromOpportunity, openOpportunityForm, getOpportunityContacts } from '../../store/OpportunityDetails/Actions';

export interface ContactProps {
  data: OpportunityContact;
}

interface Props {
  opportunityId: string;
}

export const ContactAccordian: React.FC<Props> = ({ opportunityId }) => {
  const [activeClass, setActiveClass] = React.useState('');
  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  const { contacts } = state.opportuntyDetails;
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  const openAddContactForm = () => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: true, groupName: 'add_contact', action: 'edit' }));
  };

  React.useEffect(() => {
    dispatch(getOpportunityContacts(opportunityId));
  }, []);

  return (
    <Accordion defaultActiveKey="0">
      <Card className="add-details">
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          Contacts
          {state.opportuntyDetails.editOportunity.allowEdit ? (
            <Image src={ImageConfig.ADD_BTN} alt="Add" title="Add" onClick={openAddContactForm} />
          ) : null}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <div className="container-fluid">
            <div className="row">
              {contacts.length ? (
                contacts.map((obj: OpportunityContact) => {
                  return <ContactCards data={obj} />;
                })
              ) : (
                <div className="no-data-txt"> No Contacts Found </div>
              )}
            </div>
          </div>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

export const ContactCards: React.FC<React.PropsWithChildren<ContactProps>> = ({ data }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  const [contactDetails, setContactDetails] = React.useState<models.OpportunityDetailsGroupItem[]>([]);
  const [opportunityRole, setOpportunityRole] = React.useState<string | undefined>('');
  React.useEffect(() => {
    loadContactAttributes();
  }, []);

  const loadContactAttributes = async () => {
    const response = await OpportunityDetailsApi.getContactDetails(data.contactId);
    setContactDetails(response);
    const addressObj = response.find((obj: any) => obj.attributeType === 'ADDRESS');
    // eslint-disable-next-line no-param-reassign
    data.visitingAddress = addressObj?.attributeValue;
    const roleObj = response.find((obj) => obj.attributeType === 'ROLE');
    setOpportunityRole(roleObj?.attributeValue);
  };

  const removeContact = async () => {
    const params: DeleteCustomerContactParams = {
      contactId: data.contactId,
      contactParentId: data.contactParentId,
      contactParentFile: data.contactParentFile,
    };
    dispatch(deleteContactFromOpportunity(params));
  };

  const changeRole = async (e: React.ChangeEvent<HTMLSelectElement>) => {
    const newValue = e.currentTarget.value;

    if (opportunityRole) {
      const roleObj = contactDetails.find((obj) => obj.attributeType === 'ROLE');
      const valueId = roleObj?.valueId ? roleObj.valueId : '';
      const params: models.SaveAttributeFieldParam = {
        attributeType: 'ROLE',
        attributeValue: newValue,
        valueId,
      };
      await Attributes.updateAttribute(params);
      setOpportunityRole(newValue);
    } else {
      await Attributes.addAttributes('SROMOPCH', data.contactId, 'ROLE', newValue);
      setOpportunityRole(newValue);
    }
  };

  return (
    <div className="col-md-4">
      <div className="card mb-4">
        <div className="card-body">
          <div className="name-add-sec">
            <p className="person-name-desig">
              <span className="name">{data.contactPerson}</span>
              <span className="designation">Associate - Sales</span>
            </p>
            <p className="mailid">{data.email ? data.email : '--'}</p>
            <p className="contact-num">{data.phone ? data.phone : '--'}</p>
            <p className="contact-address">{data.visitingAddress ? data.visitingAddress : '--'}</p>
          </div>
          {state.opportuntyDetails.editOportunity.allowEdit ? (
            <button type="button" className="address-card-close link-anchor-button" onClick={removeContact}>
              <Image className="card-delete" height="20" src={ImageConfig.CLOSE_BTN} />
            </button>
          ) : null}

          <div className="select-opportunity-sec">
            {state.opportuntyDetails.editOportunity.allowEdit ? (
              <div className="form-group">
                <label htmlFor="contact-role">Select Opportunity Role</label>
                <select className="form-control opportunity-dd" id="contact-role" value={opportunityRole} multiple={false} onChange={changeRole}>
                  {state.enviornmentConfigs.opportunityContactRoles.map((obj: models.DropDownValue) => {
                    return obj.fieldDescription ? (
                      <option value={obj.valueField}>
                        {obj.valueField} - {obj.fieldDescription}
                      </option>
                    ) : (
                      <option value={obj.valueField}>{obj.valueField}</option>
                    );
                  })}
                </select>
              </div>
            ) : (
              <p className="contact-address">{opportunityRole || '--'}</p>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};
