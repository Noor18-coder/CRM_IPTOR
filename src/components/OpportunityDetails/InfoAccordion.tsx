import React from 'react';
import { includes } from 'lodash';
import { Accordion, Card } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { AppState } from '../../store/store';
import ImageConfig from '../../config/ImageConfig';
import { getDateInFormat } from '../../helpers/utilities/lib';

import {
  OpportunityDetailsBasicInfo,
  AttributeField,
  IAttributesList,
  AttributeValueObject,
  OpportunityEditOptions,
  OpportunityType,
} from '../../helpers/Api/models';

interface Props {
  title: string;
  data: OpportunityDetailsBasicInfo[];
  openEditOpportunity: () => void;
}

export const InfoAccordion: React.FC<Props> = ({ title, data, openEditOpportunity }) => {
  const state: OpportunityEditOptions = useSelector((appState: AppState) => appState.opportuntyDetails.editOportunity);

  const [activeClass, setActiveClass] = React.useState('');
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };
  return (
    <Accordion defaultActiveKey="0">
      <Card>
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          {title}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <div className="accr-body-container">
            {state.allowEdit ? (
              <button type="button" className="edit-icon" onClick={openEditOpportunity}>
                <img src={ImageConfig.EDIT_ICON} alt="edit" />
              </button>
            ) : null}
            <ul className="list-inline bdy-list-item accr-list-columns">
              {data.map((obj: OpportunityDetailsBasicInfo) => {
                return (
                  <li className="list-inline-item">
                    <span>{obj.description}</span>
                    {obj.attributeValue ? `${obj.attributeValue}` : 'NA'}
                  </li>
                );
              })}
            </ul>
          </div>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

interface GroupAccordianProps {
  title: string;
  openEditForm: (groupName: string) => void;
  oppType: string;
}

export const InfoAccordionGroups: React.FC<GroupAccordianProps> = ({ title, openEditForm, oppType }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const { attributes, opportunityDefaultParams } = state.opportuntyDetails;
  const [moreInformationGroups, setMoreInformationGroups] = React.useState<IAttributesList[]>([]);
  const [activeClass, setActiveClass] = React.useState('');

  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  const setAllFields = (_opptyType: string) => {
    const groups = new Set(
      state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
        return obj.group;
      })
    );

    const recordType: OpportunityType | undefined = state.enviornmentConfigs.crmOpportunityTypes.find((obj: OpportunityType) => {
      return obj.oppRecordType === _opptyType;
    });

    const optionalFields = recordType && recordType.OPTIONAL_FIELDS ? recordType.OPTIONAL_FIELDS : [];
    const mandatoryFields = recordType && recordType.MANDATORY_FIELDS ? recordType.MANDATORY_FIELDS : [];
    const strArray = [...optionalFields, ...mandatoryFields];
    const tempAttributes: AttributeField[] = state.enviornmentConfigs.opportunityAttributes || [];

    if (strArray.length) {
      const response: IAttributesList[] = [];
      const filteredAttributes: AttributeField[] = tempAttributes.filter((obj: AttributeField) => {
        return includes(strArray, obj.attributeType);
      });

      groups.forEach((group: string) => {
        const items: AttributeField[] = filteredAttributes.filter((obj) => obj.group === group);
        if (items.length) {
          response.push({ group, items });
        }
      });

      setMoreInformationGroups(response);
    } else {
      const response: IAttributesList[] = [];
      groups.forEach((group: string) => {
        const groupAttributes: IAttributesList = { group, items: [] };
        groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group === group);
        response.push(groupAttributes);
      });
      setMoreInformationGroups(response);
    }
  };

  React.useEffect(() => {
    setAllFields(oppType);
  }, [oppType]);

  return (
    <Accordion defaultActiveKey="0">
      <Card>
        <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
          {title}
        </Accordion.Toggle>
        <Accordion.Collapse eventKey="1">
          <>
            {opportunityDefaultParams.oppRecordType && moreInformationGroups && moreInformationGroups?.length
              ? moreInformationGroups.map((key: IAttributesList) => {
                  return (
                    <DisplayGroup
                      title={key.group}
                      fields={key.items}
                      data={attributes}
                      openEditForm={openEditForm}
                      allowEdit={state.opportuntyDetails.editOportunity.allowEdit}
                    />
                  );
                })
              : null}
          </>
        </Accordion.Collapse>
      </Card>
    </Accordion>
  );
};

interface GroupData {
  title: string;
  fields: AttributeField[];
  data: AttributeValueObject[];
  openEditForm: (key: string) => void;
  allowEdit?: boolean;
}

export const DisplayGroup: React.FC<GroupData> = ({ title, fields, data, openEditForm, allowEdit }) => {
  const getValue = (obj: AttributeField) => {
    const dataObject = data.find((object: AttributeValueObject) => {
      return object.attributeType === obj.attributeType;
    });
    const field = fields.find((object: AttributeField) => {
      return object.attributeType === obj.attributeType;
    });
    const value = dataObject && dataObject.attributeValue ? dataObject.attributeValue : '';
    if (field?.valueFormatDesc === 'DATE') {
      if (value !== '') {
        const year = value.substring(0, 4);
        const month = value.substring(4, 6);
        const day = value.substring(6, 8);
        const tempDate = `${year}-${month}-${day}`;
        return getDateInFormat(new Date(tempDate));
      } else {
        return '--';
      }
    } else if (field?.valueFormatDesc === 'BOOLEAN') {
      return (
        <span className="checkbox-label">
          <label className="switch value-checkbox">
            <input type="checkbox" tabIndex={0} checked={value === 'Y'} />
            <span className="slider round disabled-checkbox">&nbsp;</span>
          </label>
        </span>
      );
    }
    return value;
  };

  return (
    <div className="more-info-group-container">
      <div className="more-info-group-name">
        {title}
        {allowEdit ? (
          <button type="button" className="edit-moreinfo" onClick={() => openEditForm(title)}>
            <img src={ImageConfig.EDIT_ICON} alt="edit" />
          </button>
        ) : null}
      </div>
      <div className="accr-body-container">
        <ul className="list-inline bdy-list-item accr-list-columns">
          {fields.map((obj: AttributeField) => {
            if (obj.uniqueRecord) {
              return (
                <li className="list-inline-item">
                  <span>{obj.description}</span>
                  {getValue(obj)}
                </li>
              );
            } else {
              const dataObj: AttributeValueObject[] = data.filter((dataObject: AttributeValueObject) => {
                return dataObject.attributeType === obj.attributeType;
              });
              if (dataObj.length) {
                return dataObj.map((elem: AttributeValueObject) => {
                  return (
                    <li className="list-inline-item">
                      <span>{obj.description}</span>
                      {elem.attributeValue}
                    </li>
                  );
                });
              } else {
                return (
                  <li className="list-inline-item">
                    <span>{obj.description}</span>
                    --
                  </li>
                );
              }
            }
          })}
        </ul>
      </div>
    </div>
  );
};

interface GroupMobileData {
  fields: AttributeField[];
  data: AttributeValueObject[];
}

export const DisplayGroupMobile: React.FC<GroupMobileData> = ({ fields, data }) => {
  const getValue = (obj: AttributeField) => {
    const dataObject = data.find((object: AttributeValueObject) => {
      return object.attributeType === obj.attributeType;
    });

    const value = dataObject && dataObject.attributeValue ? dataObject.attributeValue : '';

    if (!value) {
      return '-';
    }

    if (value && obj?.valueFormatDesc === 'DATE') {
      const year = value.substring(0, 4);
      const month = value.substring(4, 6);
      const day = value.substring(6, 8);
      return `${year}-${month}-${day}`;
    } else if (obj?.valueFormatDesc === 'BOOLEAN') {
      return value ? (value === 'Y' ? 'No' : 'Yes') : '-';
    }
    return value;
  };
  return (
    <div className="more-info-group-container">
      <div className="accr-body-container">
        <ul className="list-inline bdy-list-item">
          {fields.map((obj: AttributeField) => {
            return (
              <li className="list-inline-item">
                <span>{obj.description}</span>
                {getValue(obj)}
              </li>
            );
          })}
        </ul>
      </div>
    </div>
  );
};

export const AccordianForMobileWithGroups: React.FC<GroupAccordianProps> = ({ title, openEditForm, oppType }) => {
  const state: AppState = useSelector((appState: AppState) => appState);
  const { attributes } = state.opportuntyDetails;
  const [moreInformationGroups, setMoreInformationGroups] = React.useState<IAttributesList[]>([]);
  const [activeClass, setActiveClass] = React.useState('');
  const toggleAccordion = () => {
    setActiveClass(activeClass === '' ? 'active' : '');
  };

  // React.useEffect(() => {
  //   const groups = new Set(
  //     state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
  //       return obj.group;
  //     })
  //   );
  //   const response: IAttributesList[] = [];
  //   groups.forEach((group: string) => {
  //     const groupName = group.toLowerCase();
  //     const groupAttributes: IAttributesList = { group: groupName, items: [] };
  //     groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group.toLowerCase() === groupName);
  //     response.push(groupAttributes);
  //   });
  //   setMoreInformationGroups(response);
  // }, []);

  const setAllFields = (_opptyType: string) => {
    const groups = new Set(
      state.enviornmentConfigs.opportunityAttributes.map((obj: any) => {
        return obj.group;
      })
    );

    const recordType: OpportunityType | undefined = state.enviornmentConfigs.crmOpportunityTypes.find((obj: OpportunityType) => {
      return obj.oppRecordType === _opptyType;
    });

    const optionalFields = recordType && recordType.OPTIONAL_FIELDS ? recordType.OPTIONAL_FIELDS : [];
    const mandatoryFields = recordType && recordType.MANDATORY_FIELDS ? recordType.MANDATORY_FIELDS : [];
    const strArray = [...optionalFields, ...mandatoryFields];
    const tempAttributes: AttributeField[] = state.enviornmentConfigs.opportunityAttributes || [];

    if (strArray.length) {
      const response: IAttributesList[] = [];
      const filteredAttributes: AttributeField[] = tempAttributes.filter((obj: AttributeField) => {
        return includes(strArray, obj.attributeType);
      });

      groups.forEach((group: string) => {
        const items: AttributeField[] = filteredAttributes.filter((obj) => obj.group === group);
        if (items.length) {
          response.push({ group, items });
        }
      });

      setMoreInformationGroups(response);
    } else {
      const response: IAttributesList[] = [];
      groups.forEach((group: string) => {
        const groupAttributes: IAttributesList = { group, items: [] };
        groupAttributes.items = state.enviornmentConfigs.opportunityAttributes.filter((obj) => obj.group === group);
        response.push(groupAttributes);
      });
      setMoreInformationGroups(response);
    }
  };

  React.useEffect(() => {
    setAllFields(oppType);
  }, [oppType]);

  return (
    <div className="opp-moreinfo-sec">
      {title}
      <>
        {moreInformationGroups && moreInformationGroups.length
          ? moreInformationGroups.map((obj: IAttributesList) => {
              return (
                <Accordion defaultActiveKey="0">
                  <Card>
                    <Accordion.Toggle className={activeClass} onClick={toggleAccordion} as={Card.Link} eventKey="1">
                      {obj.group}
                      {state.opportuntyDetails.editOportunity.allowEdit ? (
                        <button type="button" className="mob-edit-icon" onClick={() => openEditForm(obj.group)}>
                          <img src={ImageConfig.EDIT_ICON} alt="edit" />
                        </button>
                      ) : null}
                    </Accordion.Toggle>
                    <Accordion.Collapse eventKey="1">
                      <DisplayGroupMobile fields={obj.items} data={attributes} />
                    </Accordion.Collapse>
                  </Card>
                </Accordion>
              );
            })
          : null}
      </>
    </div>
  );
};
