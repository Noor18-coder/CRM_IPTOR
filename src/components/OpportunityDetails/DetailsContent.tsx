import React from 'react'
import Container from 'react-bootstrap'
import Actions from './Actions'
import InfoAccordion from './InfoAccordion'
import Product from './Product'
import ProductMobile from './ProductMobile'
import Staging from './Staging'
import  {DetailsData} from '../../mocks/OpportunityDetails.mock';

export interface Data {

    data:DetailsData

}


const DetailsContent:React.FC<Data> = (props) => {
    return (
        // <!-- MAIN BODY CONTENT SECTION START -->
        <section className="main-wrapper opportunity">
        <div className="container-fluid">
            <Actions/>
            <Product data={props.data}/>
            <ProductMobile data={props.data}/>
            <Staging/>
            <InfoAccordion/>
        </div>
        </section>
        //   <!-- MAIN BODY CONTENT SECTION END -->
    )
}

export default DetailsContent
