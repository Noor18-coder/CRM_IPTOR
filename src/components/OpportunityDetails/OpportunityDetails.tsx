import React from 'react';
import { useMediaQuery } from 'react-responsive';
import { Dispatch } from 'redux';
import { useSelector, useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { Popover, Image } from 'react-bootstrap';
import { isArray } from 'lodash';
import i18n from '../../i18n';
import { AppState } from '../../store/store';
import Header from '../Shared/Header/Header';
import Footer from '../Shared/Footer/Footer';
import { InfoAccordion, InfoAccordionGroups, AccordianForMobileWithGroups } from './InfoAccordion';
import { ContactAccordian } from './ContactDetails';
import { ProductAccordian } from './ProductDetails';
import OpportunityInfo from './OpportunityInfo';
import OpportunityInfoMobile from './OpportunityInfoMobile';
import * as models from '../../helpers/Api/models';
import OpportunityDetailsApi from '../../helpers/Api/OpportunityDetailsApi';
import { NavSection } from '../Shared/DetailsNav/NavSection';
import Loader from '../Shared/Loader/Loader';
import Container from '../EditOpportunity/Container';
import ImageConfig from '../../config/ImageConfig';

import {
  getOpportunityDetails,
  openOpportunityForm,
  getOpportunityAttributes,
  setEditOpportunityErrorMessage,
  resetOpportunityDetails,
  setOpportunityEditSuccess,
} from '../../store/OpportunityDetails/Actions';
import { getCurrencySymbol, getDateInFormat } from '../../helpers/utilities/lib';
import { AreaInfo } from '../../helpers/Api/models';

export const OpportunityDetails: React.FC<any> = (props: any) => {
  const isMobile = useMediaQuery({ maxWidth: 767.98 });
  const isTablet = useMediaQuery({ minWidth: 768, maxWidth: 991 });
  const isDesktop = useMediaQuery({ minWidth: 992 });

  const state: AppState = useSelector((appState: AppState) => appState);
  const dispatch: Dispatch<any> = useDispatch();
  // eslint-disable-next-line react/destructuring-assignment
  const opportunityId = props.location.state.oppid;

  const getName = (str: string) => {
    const userObj = state.users.users.find((obj) => obj.user === str);
    return userObj?.description ? userObj?.description : '';
  };

  const getArea = (str: string) => {
    const areaObj: AreaInfo | undefined = state.enviornmentConfigs.crmAreaInfo.find((obj: AreaInfo) => {
      return obj.area === str;
    });
    return areaObj && areaObj?.description ? areaObj?.description : '';
  };

  const popover = () => {
    return (
      <Popover id="popover-basic" className="tool-tip">
        <Popover.Content>
          <button type="button" className="link-assign-to" onClick={assignOpportunity}>
            <Image className="assign-to-image" src={ImageConfig.ASSIGN_TO} alt="Iptor" title="Iptor" />
            <span className="assign-to-text">Assign To</span>
          </button>
        </Popover.Content>
        {state.auth.user.role?.toLowerCase() === 'admin' ? (
          <Popover.Content>
            <button type="button" className="link-assign-to" onClick={deleteOpportunity}>
              <Image className="popover-icon" src={ImageConfig.DELETE_ICON} alt="Iptor" title="Iptor" />
              <span className="text-link">Delete</span>
            </button>
          </Popover.Content>
        ) : null}
      </Popover>
    );
  };

  React.useEffect(() => {
    dispatch(resetOpportunityDetails());
    dispatch(getOpportunityDetails(opportunityId));
    dispatch(getOpportunityAttributes(opportunityId));
    dispatch(openOpportunityForm({ open: false }));
    document.body.classList.remove('body-scroll-hidden');
  }, []);

  const getBasicInfo = (basicInfo: models.OpportunityDetailsDefault) => {
    const data: models.OpportunityDetailsBasicInfo[] = [];

    Object.entries(basicInfo).forEach(([key, value]) => {
      switch (key) {
        case 'opportunityId':
          data.push({ description: 'Opp Number', attributeValue: value });
          break;
        case 'desc':
          data.push({ description: 'Opp Name', attributeValue: value });
          break;
        case 'userId':
          data.push({ description: 'Owner', attributeValue: value ? getName(value) : '' });
          break;
        case 'stage':
          data.push({ description: 'Stage', attributeValue: value });
          break;
        case 'currency':
          data.push({ description: 'Currency', attributeValue: value });
          break;
        case 'endDate':
          // eslint-disable-next-line no-case-declarations
          const datestr = getDateInFormat(new Date(value));
          data.push({ description: 'Close Date', attributeValue: datestr });
          break;
        case 'oppRecordType':
          data.push({ description: 'Opportunity Type', attributeValue: value });
          break;
        case 'estimatedValue':
          data.push({ description: 'Opp Value', attributeValue: `${getEstimatedValue(basicInfo)}` });
          break;
        case 'area':
          data.push({ description: 'Area', attributeValue: `${getArea(value)}` });
          break;
        case 'forecastCategory':
          data.push({ description: 'Forecast Category', attributeValue: value });
          break;
        default:
          break;
      }
    });
    return data;
  };

  const getEstimatedValue = (basicInfo: models.OpportunityDetailsDefault) => {
    // eslint-disable-next-line max-len
    return `${
      state.enviornmentConfigs.defaultOpprtunityInfo.currencyLDA && getCurrencySymbol(state.enviornmentConfigs.defaultOpprtunityInfo.currencyLDA)
    } ${basicInfo.estimatedValueSys ? Math.round(basicInfo.estimatedValueSys) : ''}
    (${basicInfo.currency ? getCurrencySymbol(basicInfo.currency) : ''} ${basicInfo.estimatedValue ? Math.round(basicInfo.estimatedValue) : ''} )`;
  };

  const history = useHistory();
  const backToOpportunityList = () => {
    dispatch(resetOpportunityDetails());
    history.goBack();
  };

  const openEditForm = (groupName: string) => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: true, groupName, action: 'edit' }));
  };

  const openOpportunityBasicEdit = () => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: true, groupName: 'opportunity_defaults', action: 'edit' }));
  };

  const deleteOpportunity = async () => {
    try {
      const response: models.OpportunityDeleteResponse = await OpportunityDetailsApi.opportunityDelete(opportunityId);
      if (response && response.messages && isArray(response.messages) && response.messages[0] && response.messages[0].text) {
        dispatch(setEditOpportunityErrorMessage(response.messages[0].text));
      } else {
        dispatch(setOpportunityEditSuccess({ success: true, error: i18n.t('opportunityDeletedSuccess') }));
        setTimeout(() => {
          backToOpportunityList();
        }, 2000);
      }
    } catch (error) {
      console.error(error);
    }
  };

  const assignOpportunity = () => {
    document.body.classList.add('body-scroll-hidden');
    dispatch(openOpportunityForm({ open: true, groupName: 'assign_opportunity', action: 'edit' }));
  };

  const hideErrorMessage = () => {
    dispatch(setEditOpportunityErrorMessage(''));
  };

  return (
    <>
      <Header page={1} />
      {state.opportuntyDetails.loader ? <Loader component="opportunity" /> : null}
      <section className="main-wrapper opportunity">
        {(state.opportuntyDetails.editOportunity.open === false && state.opportuntyDetails.editOportunity.error) ||
        (state.opportuntyDetails.editOportunity.success && state.opportuntyDetails.editOportunity.error !== '') ? (
          <div className="iptor-alert">
            <div className="alert-wrapper">
              <div role="alert" className="alert alert-danger">
                <button className="btn alert-close" type="button" onClick={hideErrorMessage}>
                  <img src={ImageConfig.CLOSE_BTN} alt="close" />
                </button>
                {`${state.opportuntyDetails.editOportunity.error}`}
              </div>
            </div>
          </div>
        ) : null}
        <div className="container-fluid">
          <NavSection backToOpportunityList={backToOpportunityList} popover={popover} />
          {state.opportuntyDetails.opportunityDefaultParams.opportunityId ? (
            <>
              <OpportunityInfo />
              <OpportunityInfoMobile popover={popover} />
              <section className="sec-info-accordion">
                <InfoAccordion
                  title="Basics"
                  data={getBasicInfo(state.opportuntyDetails.opportunityDefaultParams)}
                  openEditOpportunity={openOpportunityBasicEdit}
                />

                {isDesktop ? (
                  <InfoAccordionGroups
                    title="More Information"
                    openEditForm={openEditForm}
                    oppType={state.opportuntyDetails.opportunityDefaultParams.oppRecordType}
                  />
                ) : null}
                {isTablet || isMobile ? (
                  <AccordianForMobileWithGroups
                    title="More Information"
                    openEditForm={openEditForm}
                    oppType={state.opportuntyDetails.opportunityDefaultParams.oppRecordType}
                  />
                ) : null}
                <ProductAccordian opportunityId={opportunityId} />
                <ContactAccordian opportunityId={opportunityId} />
              </section>
            </>
          ) : null}
        </div>
      </section>
      <Footer />
      <Container />
    </>
  );
};
