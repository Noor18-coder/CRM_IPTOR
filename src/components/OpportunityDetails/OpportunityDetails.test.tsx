import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { OpportunityDetails } from './OpportunityDetails';
import StoreMock from '../../mocks/Store.mock';

const store = StoreMock.createInitialStore();
const location = { state: { oppid: '' } };

describe('[OpportunityDetails] OpportunityDetails', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <OpportunityDetails location={location} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
