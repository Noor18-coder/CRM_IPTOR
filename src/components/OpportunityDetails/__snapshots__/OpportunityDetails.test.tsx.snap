// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[OpportunityDetails] OpportunityDetails should renders correctly 1`] = `
Array [
  <nav
    className="site-navbar navbar navbar-default bg-black"
    role="navigation"
  >
    <div
      className="navbar-container container-fluid"
    >
      <div
        className="navbar-header"
      >
        <div
          className="navbar-brand navbar-brand-center navbar-sec-logo"
        >
          <img
            alt="..."
            className="navbar-brand-logo"
            src="iptor_icon.svg"
            title="iptor"
          />
        </div>
        <div
          className="navbar-brand navbar-brand-center navbar-sec-flag"
        >
          <span
            className="navbar-brand-text hidden-xs-down"
          >
             
            
          </span>
        </div>
      </div>
      <ul
        className="nav navbar-toolbar navbar-right navbar-toolbar-right"
      >
        <li
          className="nav-item"
        >
          <a
            aria-expanded="false"
            className="nav-link navbar-avatar"
            data-animation="scale-up"
            href="/"
            onClick={[Function]}
            role="button"
          >
            <span
              className="avatar avatar-online"
            >
              <img
                alt="..."
                src="dashboard_icon.svg"
              />
              <i>
                Dashboard
              </i>
            </span>
          </a>
        </li>
        <li
          className="nav-item active"
        >
          <a
            aria-expanded="false"
            className="nav-link navbar-avatar"
            data-animation="scale-up"
            href="/opportunities"
            onClick={[Function]}
            role="button"
          >
            <span
              className="avatar avatar-online"
            >
              <img
                alt="..."
                src="nav-oppty-active-icon.svg"
              />
              <i>
                Opportunities
              </i>
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-expanded="false"
            className="nav-link navbar-avatar"
            data-animation="scale-up"
            href="/customers"
            role="button"
          >
            <span
              className="avatar avatar-online"
            >
              <img
                alt="..."
                src="nav-customers-icon.svg"
              />
              <i>
                Customers
              </i>
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-expanded="false"
            className="nav-link navbar-avatar"
            data-animation="scale-up"
            href="/reports"
            role="button"
          >
            <span
              className="avatar avatar-online"
            >
              <img
                alt="..."
                src="nav-report-icon.svg"
              />
              <i>
                Reports
              </i>
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-expanded="false"
            className="nav-link navbar-avatar header-notification"
            data-animation="scale-up"
            href="/notifications"
            role="button"
          >
            <span
              className="avatar avatar-online"
            >
              <img
                alt="..."
                src="nav-notifications-icon.svg"
              />
              <i>
                Notifications
              </i>
            </span>
          </a>
        </li>
      </ul>
      <ul
        className="nav navbar-toolbar"
      >
        <button
          className="logout"
          onClick={[Function]}
          type="button"
        >
          
        </button>
      </ul>
    </div>
  </nav>,
  <section
    className="main-wrapper opportunity"
  >
    <div
      className="container-fluid"
    >
      <section
        className="d-flex justify-content-between pagelvl-actionrow"
      >
        <div
          className="lft-actionitem"
        >
          <img
            alt="Back"
            className="action-icon"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="presentation"
            src="left-arrow.svg"
            title="Back"
          />
        </div>
        <div
          className="rgt-actionitem"
        >
          <ul
            className="list-inline "
          />
        </div>
      </section>
    </div>
  </section>,
  <footer>
    <p>
      <img
        alt="Iptor"
        src="iptor-logo-orange.svg"
        title="Iptor"
      />
       Â© All Content Copyright 2021
       
    </p>
  </footer>,
]
`;
